src/java.base/share/classes/sun/security/ssl/Finished.java

```diff
@@ -846,6 +846,11 @@ private byte[] onProduceFinished(ServerHandshakeContext shc,
 
                 // update the context for the following key derivation
                 shc.handshakeKeyDerivation = secretKD;
+                if (shc.sslConfig.isQuic) {
+                    QuicTLSEngineImpl engine =
+                            (QuicTLSEngineImpl) shc.conContext.transport;
+                    engine.deriveOneRTTKeys();
```

Comment:
We should not derive the server's 1RTT read keys before processing the client's Finished message.

Also, we could skip calculating the SSL WriteCipher when QUIC is in use. Also, we're calculating the baseWriteSecret twice (deriveOneRTTKeys calculates the same secret)